File: positiveMidElement.py
def positiveMidElement(n):
    a = []
    for k in n:
        if k > 0:
            a.append(k)
    return a[abs((len(a)-1) // 2)]


n = int(input())
t = []
while(n > 0):
    t.append(int(input()))
    n -= 1
print(positiveMidElement(t))

File: positiveMidElement.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n, k = 0;
    cin >> n;
    int arr[n];
    int tarr[n];

    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    for (int j : arr)
    {
        if (j > 0)
            tarr[k++] = j;
    }
    cout << tarr[abs((k - 1) / 2)];

    return 0;
}
